### PR TITLE
Add Catch problems early (shift left) pattern

### LINKED ISSUE #244: Catch problems early (Shift quality left) pattern
Our engineering patterns are example solutions for addressing problems, meeting standards and living our engineering principles. They should make it easier for engineers to do their jobs. We don't have a standard way of writing patterns as we like to be a bit more flexible with their structure. Please provide us with some information on what you are proposing below

**What is the pattern you are suggesting?**
Catch problems early (shift quality left)

**What would be the benefit to the Home Office of following this pattern**
Why this is important
It's important to catch problems early for the following reasons:

Any problems can be addressed more quickly.
Tighter feedback loop. Allows problems with changes to be fixed while it is still fresh in the developer's head.
Reduces development and test effort to deliver a change, as issues can be caught before a formal test phase.
Problems can be caught before they can impact on your users, which will increase your users' confidence in your service.

Reduces the risk of adverse changes entering the product.
Overall; this allows for faster iteration, whilst maintaining the quality of your service.

**How do people implement this pattern**
You should:

Enable developers to test their changes in a local development environment
Developers should be able to spin up a copy of the environment on their local machine
Define staging environments, the testing to be done in those environments, and the progression criteria
Have staging environments that match the production environment as closely as possible
Test changes in a CI pipeline before they reach a staging environment (for example, run unit & functional tests)
Test changes in staging environments before they reach production (for example, run system-wide and integration tests)

**Additional information**
Add any other information you think would be useful here

- [x] I have looked through our existing patterns and think this is not covered elsewhere
- [x] I have checked through the open issues on the repository and this pattern has not already been suggested

### PR DESCRIPTION
**Pull request by robertdeniszczyc2**
Will close #244 

# Content change 
I can confirm:
- [x] Content does not include any code or configuration changes (excluding frontmatter information)
- [x] Content meets the content standards
e.g. [Writing a principle](https://ho-cto.github.io/engineering-guidance-and-standards/docs/standards/writing-a-principle/) and [Writing a standard](https://ho-cto.github.io/engineering-guidance-and-standards/docs/standards/writing-a-standard/)
- [x] Content is suitable to open source, i.e.:
    - Content does not relate to unreleased gov policy
    - Content does not refer to anti-fraud mechanisms
    - Content does not include sensitive business logic
- [x] Last updated date for content is correct

### CODE REVIEW

File: docs/patterns/catch-problems-early.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+- Reduces development and test effort to deliver a change, as issues can be caught before a formal test phase
+- Problems can be caught before they can impact on your users, which will increase your users' confidence in your service
+- Reduces the risk of adverse changes entering the product
+
+Overall this allows for faster iteration, whilst maintaining the quality of your service.
+
+---
+
+## Solution
+
```
Comment (robertdeniszczyc2) [inline docs/patterns/catch-problems-early.md:23]: Add a line in here to highlight this list isn't exhaustive

File: docs/patterns/catch-problems-early.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
```
Comment (robertdeniszczyc2) [inline docs/patterns/catch-problems-early.md:1]: Rename file to `catch-defects-early` to be more specific in the problems we are discussing here are software problems/defects

File: docs/patterns/catch-problems-early.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+layout: pattern
+order: 1
+title: Catch problems early (shift quality left)
```
Comment (robertdeniszczyc2) [inline docs/patterns/catch-problems-early.md:4]:
```suggestion
title: Catch defects early (shift quality left)
```

File: docs/patterns/catch-problems-early.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+tags:
+  - Build, release and deploy
+  - Ways of working
+---
+
+Catching problems early means that:
+- Any problems can be addressed more quickly
+- Tighter feedback loop. Allows problems to be fixed while the change context is still fresh in the developer's head
+- Reduces development and test effort to deliver a change, as issues can be caught before a formal test phase
+- Problems can be caught before they can impact on your users, which will increase your users' confidence in your service
```
Comment (robertdeniszczyc2) [inline docs/patterns/catch-problems-early.md:15]:
```suggestion
Catching defects early means that:
- Any defects can be addressed more quickly
- Tighter feedback loop. Allows defects to be fixed while the change context is still fresh in the developer's head
- Reduces development and test effort to deliver a change, as issues can be caught before a formal test phase
- Defects can be caught before they can impact on your users, which will increase your users' confidence in your service
```

File: docs/patterns/catch-problems-early.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+- Test changes in a Continuous Integration pipeline before they reach a staging environment (for example, run unit & functional tests)
+- Test changes in staging environments before they reach production (for example, run system-wide and integration tests)
+
+---
+
+## Considerations
+
+- Be wary of manual tasks going to the left as this can slow down delivery
+- Also consider moving other activities to the left, for example secure by design activities
+- Moving quality control to the left should not be considered as a substitute to observability in production - use shift right activities to catch problems quickly in production
```
Comment (robertdeniszczyc2) [inline docs/patterns/catch-problems-early.md:37]:
```suggestion
- Moving quality control to the left should not be considered as a substitute to observability in production - use shift right activities to catch defects quickly in production
```

**Review by edhamiltonHO — APPROVED** [review]
LGTM

**Review by keithkennedyHO — APPROVED** [review]

**Pull request merged**